images/encryption/payload_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package encryption
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd/diff"
+	"github.com/gogo/protobuf/types"
+)
+
+func TestInit(t *testing.T) {
+	if !processorPayloadsUseGogo {
```

Comment:
should this be testing against if it is set - only if the precondition is met? I’m not too familiar with the new any types

Reply:
Right now, because go.mod locks our dependencies and our containerd is 1.6, processorPayloadsUseGogo is always true.

Reply:
https://github.com/containerd/containerd/pull/6832 is running processorPayloadsUseGogo=false case.